### PR TITLE
Add cylinder and cone primitives

### PR DESCRIPTION
**Pull request by HenryWConklin**
For #60

### CODE REVIEW

File: blackjack_engine/src/mesh/halfedge/primitives.rs [hunk 1]
```diff
@@ -260,6 +263,80 @@ impl Polygon {
...
+impl Cone {
+    pub fn build(
+        center: Vec3,
+        top_radius: f32,
+        bottom_radius: f32,
+        height: f32,
+        num_vertices: usize,
+    ) -> HalfEdgeMesh {
+        // Not using an epsilon, may be useful for to keep multiple verts at the point using small values.
+        if top_radius.abs() <= 1e-5 {
```
Comment (setzer22) [inline blackjack_engine/src/mesh/halfedge/primitives.rs:276]: I see what you mean, but I like this version. If some user wants the version with multiple vertices at the same point for whatever reason, they can do that with an extra node (`Edit Geometry` and scale to zero, for instance).

File: blackjack_engine/src/mesh/halfedge/primitives.rs [hunk 2]
```diff
@@ -312,3 +414,22 @@ mod lua_api {
         Polygon::build_from_points(LVec3::cast_vector(points))
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    #[test]
```
Comment (setzer22) [inline blackjack_engine/src/mesh/halfedge/primitives.rs:421]: Nice! This reminds me I should add more tests in general :sweat_smile:

**Review by setzer22 — APPROVED** [review]
Looking great! Thanks a lot :smile:  I left a couple comments but nothing you need to address. Will merge once CI passes.

**Comment by setzer22** [comment]
image

File: blackjack_engine/src/mesh/halfedge/primitives.rs (hunk 1 above)
Comment (HenryWConklin) [reply blackjack_engine/src/mesh/halfedge/primitives.rs:276]: Oops forgot to update that comment. I had it as `== 0` but scrolling the value up and down in the editor wasn't going exactly back to zero so I switched it to use the epsilon.

**Pull request merged**